covid19-dashboard_NEW/main.py

```diff
@@ -0,0 +1,360 @@
+                # Store the data for the geoId.
+                name = response[geo_id]['out'][0]['value']
+                states[geo_id] = (name, 'country/USA', "State")
+
+    # Get US County data.
+    response: dict = send_request(DC_SERVER + "node/places-in", {
+        "dcids": state_dcids,
+        "placeType": "County"
+    },
+                                  api_key=API_KEY)
+
+    # Keep track of what county each state belongs to.
+    county_to_state: dict = {}
+    for value in response:
+        if 'place' in value and 'dcid' in value:
+            county_geo_id: str = value['place']
+            belongs_to_geo_id: str = value['dcid']
+            county_to_state[county_geo_id]: str = belongs_to_geo_id
+
+    # Get the name of all counties
+    response: dict = send_request(DC_SERVER + "node/property-values", {
+        "dcids": list(county_to_state.keys()),
+        "property": "name"
+    },
+                                  api_key=API_KEY)
+
+    # Store all the counties.
+    counties: dict = {}
+
+    # If param return_counties == True, get county data.
+    if return_counties:
+        for geo_id in response:
+            if 'out' in response[geo_id] and response[geo_id]['out']:
+                if 'value' in response[geo_id]['out'][0]:
+                    name: str = response[geo_id]['out'][0]['value']
+                    # Get rid of trailing strings in names such as County.
+                    for region_type in REGION_TYPES:
+                        name: str = name.replace(" " + region_type, "")
+                    # The geoId where this region belongs to.
+                    belongs_to: str = county_to_state[geo_id]
+                    counties[geo_id] = (name, belongs_to, "County")
+
+        # NYT exceptions.
+        counties['geoId/3651000'] = ("New York City", "geoId/36", "County")
+        counties['geoId/2938000'] = ("Kansas City", "geoId/29", "County")
+
+    return {**states, **counties}
+
+
+def get_stats_by_date(place: list, stats_var: str) -> dict:
+    """
+    Queries and returns the data in the appropriate format.
+    :param place: list of dcids to get data for
+    :param stats_var: the Data Commons stats_var to query data for
+    :return: dictionary where every day contains several regions,
+    and those regions contain an int of cases/deaths
+    """
+    response: dict = send_request(DC_SERVER + "bulk/stats", {
+        "place": place,
+        "stats_var": stats_var
+    },
+                                  api_key=API_KEY)
+
+    output: defaultdict = defaultdict(dict)
+
+    # Store the data in output as dates->dcid->value
+    for dcid in response:
+        if not response[dcid]:
+            continue
+        if "data" not in response[dcid]:
+            continue
+        data: dict = response[dcid]["data"]
+        for date in data:
+            output[dcid][date] = data[date]
+    return output
+
+
+def local_cache(data, minutes=100):
```

Comment:
i think a better name for this is add_browser_cache_headers_to_response or something to that effect. local_cache is too vague.